src/queue.cpp

```diff
@@ -915,6 +958,41 @@ cvk_command_kernel::build_batchable_inner(cvk_command_buffer& command_buffer) {
         return CL_OUT_OF_RESOURCES;
     }
 
+    // Setup printf buffer descriptor if needed
+    if (m_kernel->program()->uses_printf()) {
+        // Create and initialize the printf buffer
+        auto buffer = m_queue->get_or_create_printf_buffer();
+        auto err = m_queue->reset_printf_buffer();
```

Comment:
This means we can be writing to the printf buffer concurrently to executing kernels (or maybe even the post-action code) with no protection whatsoever.